File: src/TinRoll.Logic/Mapper/UserMapper.cs
using TinRoll.Data;
using TinRoll.Shared;

namespace TinRoll.Logic.Mapper
{
    public static class UserMapper
    {
        public static UserDto ToDto(User user)
        {
            return new UserDto
            {
                Id = user.Id,
                Email = user.Email,
                Username = user.Username,
                //Questions = user.Questions,
                CreatedDate = user.CreatedDate,
                UpdatedDate = user.UpdatedDate
            };
        }
        public static User ToDb(UserDto userDto)
        {
            return new User
            {
                Id = userDto.Id,
                Email = userDto.Email,
                Username = userDto.Username,
                CreatedDate = userDto.CreatedDate,
                UpdatedDate = userDto.UpdatedDate
            };
        }
    }
}

File: src/TinRoll.Logic/Mapper/UserMapper.fs
module TinRoll.Logic.Mapper.UserMapper

open TinRoll.Data
open TinRoll.Shared
open System.Collections

let ToDto (db: User) =
    UserDto(Id= db.Id, Email = db.Email, Username = db.Username, CreatedDate = db.CreatedDate, UpdatedDate = db.UpdatedDate)

let ToDb (dto: UserDto) =
    User(dto.Id, dto.Email, dto.Username, new Generic.List<Question>(), dto.CreatedDate, dto.CreatedDate)

